### DISCUSSION TITLE
Missing relationship fields with many in Database API find query results

### DISCUSSION THREAD

**Discussion by MurzNN** [keystonejs/keystone#7313] [Questions] [answered]
1. Start the `examples/blog` project, fill with some data.
2. In `onConnect` function write two queries:
```
      console.log(await context.db.Post.findMany({ take: 1 }));
      console.log(await context.db.Author.findMany({ take: 1 }));
```
3. See in console that in `Author` list item the `posts` relationship field is missing.
But in `Post` the `author` relationship field is present.

Will be good to describe what rule does depend on this, seems it is simple like "only relationships with many = false"?
And what about update functions - can we fill relationships with many=true or not?

So please add this nuance to the documentation here https://keystonejs.com/docs/apis/db-items

And maybe there is some option exists to load item with many relationship fields?

**Comment by MurzNN** [comment]
In `context.prisma` we can require needed relationship fields individually via nested reads: https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#nested-reads
```js
      const result = await context.prisma.author.findMany({
        take: 2,
        include: {
          posts: {
            select: {
              id: true,
            },
          },
        },
      });
      console.log(result[0]);
```

So maybe let expose this Prisma API to Database API too?

**Comment by MurzNN** [comment]
Together with adding `include` and `select` for relationship fields, we will also give the ability to [query only specific fields from database](https://www.prisma.io/docs/concepts/components/prisma-client/select-fields#select-specific-fields) like this:
```js
      const result = await context.db.Author.findMany({
        take: 2,
        select: {
          name: true,
        },
        include: {
          posts: {
            select: {
              id: true,
            },
          },
        },
      });
```

**Comment by MurzNN** [comment]
Seems we can simply add `include` and `select` arguments to current `findMany()` and related functions here:
https://github.com/keystonejs/keystone/blob/4d7483d9bbd6a4213f7100e6624367f6a0953b68/packages/keystone/src/lib/core/queries/resolvers.ts#L109-L110
and pass them to `runWithPrisma()` directly, something like this: https://github.com/keystonejs/keystone/pull/7204
Is this right way to implement that?

**Comment by dcousens** [comment, marked as answer]
Hi @MurzNN,

The `context.db` API uses the GraphQL resolvers internally, but typically does not resolve relationships.

The reason you _sometimes_ find a relationship field in the returned result is an implementation detail of how relationships are stored within the underlying database;  and this may change depending on the relationship configuration and the details of the underlying database.

If you need relationships, I'd recommend you use the `context.query` API instead.

**Reply by MurzNN** [reply]
The problem with Query API is that I can't query all item fields without manually describing each item in query, only DB API gives this.

So why we can't simply add `include` argument to DB API?
